src/tools/cephfs/Dumper.cc

```diff
@@ -125,11 +115,39 @@ int Dumper::dump(const char *dump_file)
       ::close(fd);
       return r;
     }
-    r = bl.write_fd(fd);
-    if (r) {
-      derr << "Error " << r << " (" << cpp_strerror(r) << ") writing journal file" << dendl;
-      ::close(fd);
-      return r;
+
+
+    // Read and write 32MB chunks.  Slower than it could be because we're not
+    // streaming, but that's okay because this is just a debug/disaster tool.
+    const uint32_t chunk_size = 32 * 1024 * 1024;
+
+    for (uint64_t pos = start; pos < start + len; pos += chunk_size) {
+      bufferlist bl;
+      dout(10) << "Reading at pos=0x" << std::hex << pos << std::dec << dendl;
+
+      const uint32_t read_size = MIN(chunk_size, end - pos);
+
+      C_SaferCond cond;
+      lock.Lock();
+      filer.read(ino, &journaler.get_layout(), CEPH_NOSNAP,
+                 pos, read_size, &bl, 0, &cond);
+      lock.Unlock();
+      r = cond.wait();
+      if (r < 0) {
+        derr << "Error " << r << " (" << cpp_strerror(r) << ") reading "
+                "journal at offset 0x" << std::hex << pos << std::dec << dendl;
```

Comment:
Actually I lie, previously the return value of Filer::read was just never checked!  So the old behaviour on errors would be to write whatever was in the bufferlist after an errored read, and return successfully!  Either way, this patch is about the chunking rather than adding in resilience